File: TN-POS-SYS/TN-SYS-API/src/internal/model/m_db_guid_itt.go
// 🇻🇳 Base Item Model với GUID
// 🇺🇸 Base Item Model with GUID
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
)

// M_Db_Guid_Itm Base item với GUID ID
type M_Db_Guid_Itm struct {
	QID string `json:"q_id" gorm:"column:q_id;type:uuid;primaryKey;default:gen_random_uuid()"`
	M_Db_Ett
}

// BeforeCreate hook để generate UUID nếu chưa có
// Note: Cần implement GORM hook interface nếu muốn tự động
func (m *M_Db_Guid_Itm) BeforeCreate(tx *gorm.DB) error {
	if m.QID == "" {
		m.QID = uuid.New().String()
	}
	return nil
}

File: TN-POS-SYS/TN-SYS-WEB/src/core/models/m_db_guid_itt.ts
// 🇻🇳 Model cơ sở với GUID Identity
// 🇺🇸 Base model with GUID Identity
import type { M_Db_Itm } from './m_db_itt';

export interface M_Db_Guid_Itm extends M_Db_Itm<string> {
  // q_id: string (inherited from M_Db_Itm<string>)
}

